src/runtime/plugins/critical.server.ts

```diff
@@ -1,40 +1,11 @@
-import type { Browser, parseUserAgent } from 'detect-browser-es'
-import type {
-  ResolvedHttpClientHintsOptions,
-  CriticalInfo,
-  CriticalClientHintsConfiguration,
-} from '../shared-types/types'
+import type { parseUserAgent } from 'detect-browser-es'
+import { CriticalHintsHeaders, extractCriticalHints } from '../utils/critical'
+import type { ResolvedHttpClientHintsOptions } from '../shared-types/types'
 import { useHttpClientHintsState } from './state'
-import { lookupHeader, writeClientHintHeaders, writeHeaders } from './headers'
-import { browserFeatureAvailable } from './features'
-import {
-  defineNuxtPlugin,
-  useCookie,
-  useRuntimeConfig,
-  useRequestHeaders,
-} from '#imports'
+import { writeHeaders } from './headers'
+import { defineNuxtPlugin, useCookie, useRequestHeaders, useRuntimeConfig } from '#imports'
```

Comment:
replace `useRuntimeConfig` with `useAppConfig` in all plugins and remove the registration from the configuration